### PR TITLE
[hotfix] fix ddp for unit test test_gpt2

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

File: tests/test_tensor/test_gpt2.py [hunk 1]
```diff
@@ -50,33 +47,39 @@ def check_grad_equal(model, torch_model, pg: ProcessGroup):
...
     check_param_equal(model, torch_model, pg)
-    model.train()
-    torch_model.train()
-    torch.distributed.barrier()
 
+    # close the dropout in eval mode
+    model.eval()
+    torch_model.eval()
+
+    torch.distributed.barrier()
```
Comment (feifeibear) [inline tests/test_tensor/test_gpt2.py:82]: set_seed(pg.tp_local_rank())
add it here

File: tests/test_tensor/test_model.py [hunk 2]
```diff
@@ -296,8 +299,9 @@ def _run_pretrain_load():
 
 def run_model_dist(rank, world_size, port):
     colossalai.launch(config={}, rank=rank, world_size=world_size, host='localhost', port=port, backend='nccl')
-    for name in ['bert', 'simple_net']:
-        run_1d_row_tp(name)
+    # Comment below test for speed consideration
+    # for name in ['bert', 'simple_net']:
+    #     run_1d_row_tp(name)
```
[the remark this answers is not in the document]
Comment (1SAA) [reply tests/test_tensor/test_model.py:304]: I think hybrid test is enough  for normal situations.

File: tests/test_tensor/test_gpt2.py (hunk 1 above)
Comment (1SAA) [reply tests/test_tensor/test_gpt2.py:82]: done

**Comment by 1SAA** [comment]
image

**Pull request merged**